superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -125,8 +125,16 @@ class VisualizeModal extends React.PureComponent {
     this.setState({ columns }, this.validate);
   }
   render() {
-    if (!(this.props.query)) {
-      return <div />;
+    if (!(this.props.query) || !(this.props.query.results)) {
+      return (
+        <div className="VisualizeModal">
+          <Modal show={this.props.show} onHide={this.props.onHide}>
+            <Modal.Body>
+              No results available for this query
```

Comment:
ah, i see what's happening now, i missed reading about how we flush the query results except for the latest query.